Jenkinsfile

```diff
@@ -23,17 +43,80 @@ node('us-east-1 && ubuntu && docker && !gpu') {
+                    while (status != 'passed') {
+                        println "Waiting for the instance to fully Boot up...."
+                        sleep(15)
+                        status = sh(returnStdout: true, script: """
+                            aws ec2 describe-instance-status --region=us-east-1 --instance-id $INSTANCE_ID --query 'InstanceStatuses[].InstanceStatus[].Details[].Status' --output text
+                        """).trim()
+                        println "Instance Status: ${status}"
+                    }
+                }
             }
         }
-        finally {
-            stage('Clean Up') {
-                sh '''#!/bin/bash -l
-                    rm -f mozilla-release/mozconfig
-                    rm -rf mozilla-release/objdir-frontend-android
-                '''
+        docker.image("141047255820.dkr.ecr.us-east-1.amazonaws.com/${baseImageName}").inside {
+            withEnv(["IP=${ip}", "APP=${apk}", "platformName=android", "deviceName=127.0.0.1:5556"]){
+                withCredentials([file(credentialsId: 'e87da5bf-eb00-4d85-8ef9-93680abaa266', variable: 'FILE' )]){
+                    stage('Run Tests') {
+                        timeout(60) {
+                            sh'''#!/bin/bash -l
+                                set -x
+                                set -e
+                                chmod 400 $FILE
+                                ssh -v -o StrictHostKeyChecking=no -i $FILE root@$IP "setprop persist.sys.usb.config adb"
+                                ssh -v -o StrictHostKeyChecking=no -i $FILE -NL 5556:127.0.0.1:5555 root@$IP &
+                                ~/android_home/platform-tools/adb connect 127.0.0.1:5556
+                                ~/android_home/platform-tools/adb wait-for-device
+                                appium &
+                                sleep 10
+                                export app=$PWD/$APP
+                                cd autobots
+                                chmod 0755 requirements.txt
+                                virtualenv ~/venv
+                                source ~/venv/bin/activate
+                                pip install -r requirements.txt
+                                python testRunner.py || true
+                           '''
+                       }
+                    }
+                }
+            }
+            try{
+                stage('Upload APK') {
+                    archiveArtifacts allowEmptyArchive: true, artifacts: 'mozilla-release/objdir-frontend-android/dist/*.apk'
+                    archiveArtifacts allowEmptyArchive: true, artifacts: 'autobots/*.log'
+                    junit "autobots/test-reports/*.xml"
+                    zip archive: true, dir: 'autobots/screenshots', glob: '', zipFile: 'autobots/screenshots.zip'
+                }
             }
+            catch(e) {
+                print e
+                // no screenshots, no problem
```

Comment:
but this catch covers errors not only from screenshots

Reply:
Yes. Removed the comment.